dbt/adapters/snowflake/impl.py

```diff
@@ -73,6 +76,8 @@ def _get_warehouse(self) -> str:
 
     def _use_warehouse(self, warehouse: str):
         """Use the given warehouse. Quotes are never applied."""
+        kwargs = {"warehouse": warehouse}
+        warehouse = self.execute_macro(SNOWFLAKE_WAREHOUSE_MACRO_NAME, kwargs=kwargs)  # type: ignore
```

Comment:
curious if anyone has an idea as to why this causes a 
`dbt/adapters/snowflake/impl.py:80: error: Incompatible types in assignment (expression has type "AttrDict", variable has type "str")  [assignment]` error? seems same as all implementations in `dbt-core` I could find. just curious if there is better path than the ignore

Reply:
It's silly, but you can remove the type annotation from `warehouse` and `mypy` will be happy; you can also update it to `Any`. The issue is that `self.execute_macro()` lists the type for `kwargs` as `Dict[str, Any]`. `mypy` doesn't like it when you make the type more strict (`str` is a proper subset of `Any`).

Reply:
oh that's interesting iI didn't think the type value for the "warehouse" input would bother the previously defined one for `execute_macro` thank you for clearing that up @mikealfare